app/src/apdu_handler_legacy.c

```diff
@@ -288,6 +288,7 @@ bool legacy_process_transfer_chunk(uint32_t rx) {
             if (items != LEGACY_TRANSFER_NUM_ITEMS) {
                 THROW(APDU_CODE_DATA_INVALID);
             }
+            tx_initialized = false;
```

Comment:
shouldn't this go inside the previous if statement, as it happens at the beginning of this function?